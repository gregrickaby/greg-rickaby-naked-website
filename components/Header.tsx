import Image from 'next/image'
import Logo from '../app/icon.png'

/**
 * Header component.
 */
export default function Header() {
  return (
    <header className="flex items-center gap-4" data-testid="header">
      <div>
        <Image
          alt="Greg Rickaby"
          className="rounded-full"
          height="120"
          src={Logo}
          width="120"
        />
      </div>
      <div className="not-prose">
        <h1 className="mb-3 text-3xl font-bold md:text-4xl">Greg Rickaby</h1>
        <p>Full-Stack Engineer / Photographer / Author</p>
      </div>
    </header>
  )
}
