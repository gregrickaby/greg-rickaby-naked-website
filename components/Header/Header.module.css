.header {
  @apply mb-16 flex items-center justify-between gap-12 p-8 text-center;
  @apply md:flex-row md:justify-between md:p-12;
}

.headerContainer {
  @apply flex flex-row items-center gap-4 lg:flex-row lg:text-left;
}

.logo {
  @apply mx-auto rounded-full shadow-lg transition-transform duration-300 ease-in-out;
  @apply hover:rotate-45 motion-reduce:hover:transform-none;
}

.brandingContainer {
  @apply flex flex-col text-left;
}

.siteTitle {
  @apply mb-1 text-2xl font-bold leading-none tracking-tighter md:text-4xl;
}

.siteDescription {
  @apply m-0 max-w-sm p-0 md:ml-1 md:text-lg;
}
