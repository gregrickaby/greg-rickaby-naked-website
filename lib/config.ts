import {
  IconBook,
  IconBrandAmazon,
  IconBrandBluesky,
  IconBrandCodepen,
  IconBrandFacebook,
  IconBrandFeedly,
  IconBrandGithub,
  IconBrandInstagram,
  IconBrandLinkedin,
  IconBrandThreads,
  IconBrandWordpress,
  IconBrandYoutube,
  IconRss
} from '@tabler/icons-react'

const config = {
  siteName: 'Greg Rickaby',
  siteDescription: 'Cameras and Code',
  siteUrl: 'https://gregrickaby.com',
  siteLogo: '/logo.webp',
  siteOgImage: '/og.webp',
  jobTitle: 'Technical Lead',
  email: 'greg@gregrickaby.com',
  intro: 'Disney Dad. Full-Stack Developer. Photographer. Author.',
  gravatarApiUrl:
    'https://en.gravatar.com/28af3e39c0a1fe4c31367c7e9a8bcac3.json',
  navigation: [
    {
      name: 'About',
      url: '/about'
    },
    {
      name: 'Blog',
      url: '/blog'
    },
    {
      name: 'Contact',
      url: '/contact'
    },
    {
      name: 'Media',
      url: '/media'
    },
    {
      name: 'Resources',
      url: '/resources'
    },
    {
      name: 'Resume',
      url: '/resume'
    }
  ],
  socials: [
    {
      name: 'Bluesky',
      url: 'https://bsky.app/profile/gregrickaby.bsky.social',
      icon: IconBrandBluesky
    },
    {
      name: 'Instagram',
      url: 'https://www.instagram.com/gregoryrickaby/',
      icon: IconBrandInstagram
    },
    {
      name: 'Threads',
      url: 'https://www.threads.net/@gregoryrickaby',
      icon: IconBrandThreads
    },
    {
      name: 'Facebook',
      url: 'https://www.facebook.com/gregrickaby',
      icon: IconBrandFacebook
    },
    {
      name: 'GitHub',
      url: 'https://github.com/gregrickaby/',
      icon: IconBrandGithub
    },
    {
      name: 'LinkedIn',
      url: 'https://linkedin.com/in/gregrickaby/',
      icon: IconBrandLinkedin
    },
    {
      name: 'WordPress',
      url: 'https://profiles.wordpress.org/gregrickaby/',
      icon: IconBrandWordpress
    },
    {
      name: 'Amazon Author',
      url: 'https://www.amazon.com/author/gregrickaby',
      icon: IconBrandAmazon
    },
    {
      name: 'Goodreads',
      url: 'https://www.goodreads.com/author/show/16999736.Greg_Rickaby',
      icon: IconBook
    },
    {
      name: 'CodePen',
      url: 'https://codepen.io/gregrickaby',
      icon: IconBrandCodepen
    },
    {
      name: 'YouTube',
      url: 'https://www.youtube.com/@GregRickaby',
      icon: IconBrandYoutube
    },
    {
      name: 'Feedly',
      url: 'https://feedly.com/i/subscription/feed%2Fhttps%3A%2F%2Fgregrickaby.com%2Ffeed.xml',
      icon: IconBrandFeedly
    },
    {
      name: 'RSS Feed',
      url: 'https://gregrickaby.com/feed.xml',
      icon: IconRss
    }
  ]
}

export default config
