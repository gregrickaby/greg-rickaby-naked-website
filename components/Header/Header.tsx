import Logo from '@/app/icon.png'
import {Navigation} from '@/components/Navigation'
import config from '@/lib/config'
import Image from 'next/image'
import Link from 'next/link'
import styles from './Header.module.css'

/**
 * Header component.
 */
export function Header() {
  return (
    <header className={styles.header} data-testid="header">
      <div className={styles.headerContainer}>
        <Link href="/">
          <Image
            alt="Greg Rickaby"
            className={styles.logo}
            height={96}
            loading="eager"
            src={Logo}
            width={96}
          />
        </Link>
        <div className={styles.brandingContainer}>
          <h1 className={styles.siteTitle}>
            <Link href="/">{config.siteName}</Link>
          </h1>
          <p className={styles.siteDescription}>{config.siteDescription}</p>
        </div>
      </div>
      <Navigation />
    </header>
  )
}
