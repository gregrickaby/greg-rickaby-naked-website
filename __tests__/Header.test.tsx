import Header from '@/components/Header'
import config from '@/lib/config'
import '@testing-library/jest-dom'
import {render} from '@testing-library/react'
import {axe} from 'jest-axe'
import {describe, expect, it} from 'vitest'

/**
 * Test suite for the Header component.
 */
describe('Header', () => {
  it('should render', () => {
    const {getByTestId} = render(<Header />)
    const header = getByTestId('header')
    expect(header).toBeInTheDocument()
  })

  it('should render the logo', () => {
    const {getByAltText} = render(<Header />)
    const logo = getByAltText(config.siteName)
    expect(logo).toBeInTheDocument
  })

  it('should render the site name', () => {
    const {getByText} = render(<Header />)
    const text = getByText(config.siteName)
    expect(text).toBeInTheDocument()
  })

  it('should render the description text', () => {
    const {getByText} = render(<Header />)
    const text = getByText(config.siteDescription)
    expect(text).toBeInTheDocument
  })

  it('should not have accessibility issues', async () => {
    const {container} = render(<Header />)
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })

  it('should match snapshot', () => {
    const {getByTestId} = render(<Header />)
    const header = getByTestId('header')
    expect(header).toMatchSnapshot()
  })
})
