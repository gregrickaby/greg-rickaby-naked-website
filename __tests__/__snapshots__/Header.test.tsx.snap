// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should match snapshot 1`] = `
<header
  class="flex items-center gap-4"
  data-testid="header"
>
  <div>
    <img
      alt="Greg Rickaby"
      class="rounded-full"
      data-nimg="1"
      decoding="async"
      height="120"
      loading="lazy"
      src="/_next/image?url=%2Fapp%2Ficon.png&w=256&q=75"
      srcset="/_next/image?url=%2Fapp%2Ficon.png&w=128&q=75 1x, /_next/image?url=%2Fapp%2Ficon.png&w=256&q=75 2x"
      style="color: transparent;"
      width="120"
    />
  </div>
  <div
    class="not-prose"
  >
    <h1
      class="mb-3 text-3xl font-bold md:text-4xl"
    >
      Greg Rickaby
    </h1>
    <p>
      Full-Stack Engineer / Photographer / Author
    </p>
  </div>
</header>
`;
